src/runtime/methodbinder.cs

```diff
@@ -335,6 +356,46 @@ internal Binding Bind(IntPtr inst, IntPtr args, IntPtr kw, MethodBase info, Meth
                     continue;
                 }
 
+                var matchedMethod = new MatchedMethod(kwargsMatched, defaultsNeeded, margs, outs, mi);
+                argMatchedMethods.Add(matchedMethod);
+            }
+            if (argMatchedMethods.Count() > 0)
+            {
+                var bestKwargMatchCount = argMatchedMethods.Max(x => x.KwargsMatched);
+                var fewestDefaultsRequired = argMatchedMethods.Where(x => x.KwargsMatched == bestKwargMatchCount).Min(x => x.DefaultsNeeded);
+
+                int bestCount = 0;
+                int bestMatchIndex = -1;
+
+                foreach (MatchedMethod testMatch in argMatchedMethods)
+                {
+                    if (testMatch.DefaultsNeeded == fewestDefaultsRequired && testMatch.KwargsMatched == bestKwargMatchCount)
+                    {
+                        bestCount++;
+                        if (bestMatchIndex == -1)
+                            bestMatchIndex = argMatchedMethods.IndexOf(testMatch);
```

Comment:
Instead of doing `IndexOf` here, which might be expensive (results in quadratic complexity over number of overloads), change the loop to be a `for(int index = 0; index < argMatchedMethods.Count; index++) { ... }` loop, and use `index`.